### PR TITLE
set config/docs for scalability

### PR DESCRIPTION
**Pull request by saraswatpuneet**
## Details
 - [x] Add scalability reference docs
 - [x] Set some defaults in `values.yaml`

### CODE REVIEW

**Review by mattheworris — APPROVED** [review]
🚢 it!

File: docs/src/Run/Scalability.md [hunk 1]
```diff
@@ -1,3 +1,169 @@
-# Scalability
+# **Scalability Guide for Frequency Gateway**
 
-Coming soon...
+This guide explains how to configure and manage scalability using Kubernetes Horizontal Pod Autoscaler (HPA) for Frequency Gateway, ensuring your services scale dynamically based on resource usage.
+
```
Comment (jeanettedepatie) [inline docs/src/Run/Scalability.md:4]: replace "Gateway, ensuring your..." with "Gateway to ensure..."

File: docs/src/Run/Scalability.md [hunk 2]
```diff
@@ -1,3 +1,169 @@
...
+  - [**Troubleshooting**](#troubleshooting)
+    - [HPA Not Scaling Pods](#hpa-not-scaling-pods)
+    - [Scaling Too Slowly or Too Aggressively](#scaling-too-slowly-or-too-aggressively)
+
+---
+
+## **Introduction**
+
+Kubernetes Horizontal Pod Autoscaler (HPA) helps scale your deployment based on real-time resource usage (such as CPU and memory). By configuring HPA for the Frequency Gateway, you ensure your services remain available and responsive under varying loads, scaling out when demand increases and scaling down when resources aren't needed.
+
```
Comment (jeanettedepatie) [inline docs/src/Run/Scalability.md:28]: replace comma after loads with an emdash (--)

File: docs/src/Run/Scalability.md [hunk 3]
```diff
@@ -1,3 +1,169 @@
...
+---
+
+## **Prerequisites**
+
+Before implementing autoscaling, ensure that:
+
+- Kubernetes metrics server is enabled and running (or another resource metrics provider).
```
Comment (jeanettedepatie) [inline docs/src/Run/Scalability.md:35]: Replace with "Kubernetes metrics server (or another resource metrics provider) is enabled and running."

File: docs/src/Run/Scalability.md [hunk 4]
```diff
@@ -1,3 +1,169 @@
...
+resources:
+  requests:
+    memory: "256Mi"
+    cpu: "100m"
+  limits:
+    memory: "512Mi"
+    cpu: "500m"
+```
+
+- **requests**: The amount of CPU and memory a pod is guaranteed.
```
Comment (jeanettedepatie) [inline docs/src/Run/Scalability.md:110]: I am not sure what this means.  The amount of CPU and memory in a pod is guaranteed?  Even if a pod can have more, the following amount is guaranteed to be available?

File: docs/src/Run/Scalability.md [hunk 5]
```diff
@@ -1,3 +1,169 @@
...
+
+```bash
+kubectl top pods
+```
+
+---
+
+## **Troubleshooting**
+
+### HPA Not Scaling Pods
```
Comment (jeanettedepatie) [inline docs/src/Run/Scalability.md:143]: I would say "HPA is Not Scaling Pods"

File: docs/src/Run/Scalability.md (hunk 4 above)
Comment (saraswatpuneet) [reply docs/src/Run/Scalability.md:110]: Yes the minimum requested is always available.

Comment (saraswatpuneet) [reply docs/src/Run/Scalability.md]: Improved the wordings here for clarity

**Pull request merged**